### DISCUSSION TITLE
Support kitty shell integration

### DISCUSSION THREAD

**Discussion by gutenye** [nushell/nushell#12065] [Ideas]
You can read [here](https://sw.kovidgoyal.net/kitty/shell-integration/#shell-integration) about what the Shell Integration in the kitty terminal emulator is.

I asked [the same question](https://github.com/kovidgoyal/kitty/discussions/7187#discussion-6316789) in the kitty repo. It needs NuShell support. What's the current status, and is there a plan to move it forward? Thanks

**Comment by fdncred** [comment]
Feel free to let us know what isn't supported. 

I'm sure we support some of it. Off the top of my head.
- We report the PWD OSC7
- We set the title OSC2
- We report prompt markers OCS133
- We support VSCodes OSC633
- We reset application mode each prompt

We don't yet support this stuff via a script. It's all in rust right now.

**Comment by duboisf** [comment]
As @fdncred said, most things are supported but it's disabled by default.

Set `$env.config.shell_integration = true` in your `config.nu` to enable it.

**Comment by gutenye** [comment]
Thanks.

**Comment by DaBuddhalint** [comment]
i encountered this issue with ```$env.config.shell_integration = true```

```
Error: nu::shell::invalid_config

  × Encountered 1 error(s) when updating config

Error: nu::shell::type_mismatch

  × Type mismatch at $env.config.shell_integration
   ╭─[/home/[user]/.config/nushell/config.nu:6:33]
 5 │ 
 6 │ $env.config.shell_integration = true
   ·                                 ──┬─
   ·                                   ╰── expected record, but got bool
 7 │ 
   ╰────
   ```

**Reply by krovuxdev** [reply]
@DaBuddhalint  `config nu -s | nu-highlight | less -R`

<img width="846" height="606" alt="imagen" src="https://github.com/user-attachments/assets/032a5840-bca9-4f1a-a915-4f0ce9f15f02" />
